llvm/include/llvm/IR/IntrinsicsRISCV.td

```diff
@@ -200,7 +203,8 @@ let TargetPrefix = "riscv" in {
         : DefaultAttrsIntrinsic<[llvm_anyvector_ty],
                     [LLVMMatchType<0>, llvm_ptr_ty,
                      llvm_anyint_ty, LLVMMatchType<1>],
-                    [NoCapture<ArgIndex<1>>, IntrReadMem]>, RISCVVIntrinsic {
+                    [NoCapture<ArgIndex<1>>, IntrReadMem, IntrArgMemOnly]>,
```

Comment:
I'm not sure if this is valid for strided or indexed load/store. The address isn't fully described by the pointer argument.

Reply:
The aarch64 backend adds `IntrArgMemOnly` for gather and scatter intrinsics, so I think it's ok to add it for strided/indexed load/store.

Reply:
And I removed it from X86 gather/scatter a few years ago.

```
commit 35f55d72f6f8895d64171a42ee0635ef8d76c61d
Author: Craig Topper <craig.topper@intel.com>
Date:   Fri Mar 1 13:02:40 2019

    [X86] Remove IntrArgMemOnly from target specific gather/scatter intrinsics
    
    IntrArgMemOnly implies that only memory pointed to by pointer typed arguments will be accessed. But these intrinsics allow you to pass null to the pointer argument and put the full address into the index argument. Other passes won't be able to understand this.
    
    A colleague found that ISPC was creating gathers like this and then dead store elimination removed some stores because it didn't understand what the gather was doing since the pointer argument was null.
    
    Differential Revision: https://reviews.llvm.org/D58805
    
    llvm-svn: 355228
```

Reply:
Your commit makes sense. I still don't understand the comment of `IntrArgMemOnly` well, it says: "but may access an unspecified amount", it seems allow to let the intrinsic access more memory.

Reply:
But still derived from the pointer, i.e. not outside the bounds of the underlying object

Reply:
Change to add only for unit stride load/store.